dspace-api/src/main/java/org/dspace/importer/external/ror/service/RorImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,266 @@
+            query.addParameter("query", queryString);
+        }
+
+        private SearchByQueryCallable(Query query) {
+            this.query = query;
+        }
+
+        @Override
+        public List<ImportRecord> call() throws Exception {
+            return search(query.getParameterAsClass("query", String.class));
+        }
+    }
+
+    /**
+     * This class is a Callable implementation to get an ADS entry using bibcode The
+     * bibcode to use can be passed through the constructor as a String or as
+     * Query's map entry, with the key "id".
+     *
+     * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+     */
+    private class SearchByIdCallable implements Callable<List<ImportRecord>> {
+        private Query query;
+
+        private SearchByIdCallable(Query query) {
+            this.query = query;
+        }
+
+        private SearchByIdCallable(String id) {
+            this.query = new Query();
+            query.addParameter("id", id);
+        }
+
+        @Override
+        public List<ImportRecord> call() throws Exception {
+            return searchById(query.getParameterAsClass("id", String.class));
+        }
+    }
+
+    /**
+     * This class is a Callable implementation to count the number of entries for an
+     * ADS query. This Callable use as query value to ADS the string queryString
+     * passed to constructor. If the object will be construct through Query.class
+     * instance, the value of the Query's map with the key "query" will be used.
+     * 
+     * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+     */
+    private class CountByQueryCallable implements Callable<Integer> {
+        private Query query;
+
+        private CountByQueryCallable(String queryString) {
+            query = new Query();
+            query.addParameter("query", queryString);
+        }
+
+        private CountByQueryCallable(Query query) {
+            this.query = query;
+        }
+
+        @Override
+        public Integer call() throws Exception {
+            return count(query.getParameterAsClass("query", String.class));
+        }
+    }
+
+    public Integer count(String query) {
+        try {
+            Map<String, Map<String, String>> params = new HashMap<String, Map<String, String>>();
+
+            URIBuilder uriBuilder = new URIBuilder(this.url);
+            uriBuilder.addParameter("query", query);
+
+            String resp = liveImportClient.executeHttpGetRequest(timeout, uriBuilder.toString(), params);
+            if (StringUtils.isEmpty(resp)) {
+                return 0;
+            }
+            JsonNode jsonNode = convertStringJsonToJsonNode(resp);
+            return jsonNode.at("/number_of_results").asInt();
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+        return 0;
+    }
+
+    private List<ImportRecord> searchById(String id) {
+
+        List<ImportRecord> adsResults = new ArrayList<>();
```

Comment:
The variable name is odd here, as this appears to refer to ADS, which is a different external service.  Small thing, but maybe change this to just be "importResults" or "results"?